docker/BuildJekyll.Dockerfile

```
@@ -0,0 +1,20 @@
+FROM ruby:3.0.0
```

For reproducibility, I'd recommend

```suggestion
FROM ruby:3.0.0 @ sha256:c74394d0a4a05a3068ca9d0c53c36d451e27773264b568ae16ae24be9ad5621d
```

or maybe

```suggestion
FROM ruby:3 @ sha256:bf1afb92b390d90815b83ebf175c243b21e6e5204e040d3ca03a2f4dce3fbf60
```

The `tag @ sha...` syntax will pin the image that you're using, so you don't get surprised when the ruby:3.0.0 tag changes. 

(Technically the sha part overrides the tag, so you could do just `FROM sha...`, but keeping the tag is nice because it documents where it came from, which lets certain tools (dependabot, renovate, etc.) help you keep track of what ruby versions your blog is compatible with in the future).

---

Did you just grab this sha from dockerhub? Thanks for the idea though!

---

oh i see you got this from the actual build log in the workflow

---

Hmm it's not letting me do this, I'm getting: failed to solve: dockerfile parse error on line 1: FROM requires either one or three arguments for:
```
FROM ruby:3.0.0 @ sha256:c74394d0a4a05a3068ca9d0c53c36d451e27773264b568ae16ae24be9ad5621d
```

---

Oh, jeez: I guess mint in 2024 still makes buildkit optional per https://stackoverflow.com/q/77046469

I guess we could dig out the differences between the first docker builder and buildkit, but, uh, for a lot of reasons I would.... not like to do that lol. 

Looks like you should apt-get install a thing, and then set `DOCKER_BUILDKIT=1` in your environment (or, IIRC there's a docker daemon setting that just always turns it on).

---

> oh i see you got this from the actual build log in the workflow

well akthsually I ran `docker pull ruby:3` and got the sha from that, but that's the worst way to do it (maybe except for all the others) because it pulls all the layers of the image first and then prints out the SHA last. This annoys me mainly because it had to look up the image ID at the beginning in order to pull it. But AFAIK if I wanted to skip the pull process I'd have to ask the registry myself, and the docker registry API story is still "go ~~fuck~~curl it yourself" last I heard.

idk maybe one day I'll write `docker lookup` or `... resolve` or something that takes tags and produces their image IDs without a pull.